org-formation/650-identity-providers/github-oidc-provider-access.njk

```diff
@@ -89,12 +89,15 @@ Resources:
             Condition:
               StringEquals:
                 token.actions.githubusercontent.com:aud: !Ref ClientIdList
-  {% if Repository.branch == '*' %}
               StringLike:
-                token.actions.githubusercontent.com:sub: !Sub repo:${GitHubOrg}/{{ Repository.name }}:{{Repository.branch }}
+                token.actions.githubusercontent.com:sub: [
+  {% if Repository.branch == '*' %}
+                  "!Sub repo:${GitHubOrg}/{{ Repository.name }}:{{Repository.branch }}",
   {% else %}
-                token.actions.githubusercontent.com:sub: !Sub repo:${GitHubOrg}/{{ Repository.name }}:ref:refs/heads/{{ Repository.branch }}
+                  "!Sub repo:${GitHubOrg}/{{ Repository.name }}:ref:refs/heads/{{ Repository.branch }}",
```

Comment:
To exactly match a branch this should go under `StringEquals` not `StringLike`.

Reply:
I don't think that would work. My understanding from the docs (https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_multi-value-conditions.html) is that the conditions operators are ANDed (and so are the keys under an operator). If I move the match on the branch to StringEquals, a push will match on StringEquals but will not match on StringLike (similarly a tag will match on StringLike but won't on StringEquals). With StringEquals only dealing with 'aud' and StringLike only dealing with 'sub', the StringEquals should match for both push and tag, and so should the StringLike (since it has both possible values). And the StringLike value without a wildcard is like a StringEquals.

Reply:
i see, it's so complicated. thanks for the info.